starcheck/src/starcheck.pl

```diff
@@ -32,35 +34,42 @@
 use Cwd qw( abs_path );
 
 use HTML::TableExtract;
-
 use Carp 'verbose';
 $SIG{ __DIE__ } = sub { Carp::confess( @_ )};
 
+my $sock = IO::Socket::INET->new(
+    LocalAddr => '', LocalPort => 0, Proto => 'tcp', Listen => 1);
+my $server_port = $sock->sockport();
+close($sock);
+
+# Generate a 16-character random string of letters and numbers that gets used
+# as a key to authenticate the client to the server.
+my $server_key = join '', map +(0..9,'a'..'z','A'..'Z')[rand 62], 1..16;
+
+# Configure the Python interface
+Ska::Starcheck::Python::set_port($server_port);
+Ska::Starcheck::Python::set_key($server_key);
+
+# Global process ID for fork that gets set in the parent process. This is
+# used to kill the forked process when the parent finishes.
+my $pid;
+
+# Start a server that can call functions in the starcheck package
+if ($pid = fork) {} else {
+    open(SERVER, "| python -m starcheck.server");
```

Comment:
After fiddling with this some more... it looks to me like this code was always forking and making a new Perl process and then from that making a new Python process that was sticking around even when the Perl process was killed at the end.  I don't actually understand why the Perl $pid process is needed instead of tracking the python process id.

Reply:
The Perl $pid is the only one that it knows. See comment below about child vs. parent context.

Reply:
Which comment?  It certainly could know the python process and I think you've moved to my suggestion.  I had these all as commits on the side but wasn't sure if just adding them (like you told me to) was a good idea without more discussion.